java9/src/main/java/module-info.java

```diff
@@ -0,0 +1,5 @@
+module io.perfmark.java9 {
+    requires io.perfmark.impl;
```

Comment:
I am not familiar with the module system, but should a `module-info.java` file be added to the `impl` package too? In that way the java 9 module system can find this required `io.perfmark.impl` correctly. I assume those Travis build fails are because of this.

Reply:
Impl is shared by the java6 and java7 modules, so it can't take on a JDK 9+ class.  

The build failure is due to the modularity plugin not constructing the module path correctly for the tests (its in the main path, but not inherited to the test path)

Reply:
I see.